docs/guides/integration-dsp-no-sdk.md

```diff
@@ -0,0 +1,128 @@
+
+The UID2 SDK for C# / .NET uses a `DecryptTokenIntoRawUid` function to perform these steps: see [BidstreamClient.cs, Line 15](https://github.com/IABTechLab/uid2-client-net/blob/6ac53b106301e431a4aada3cbfbb93f8164ff7be/src/UID2.Client/BidstreamClient.cs#L15).
+
+### Check the Token Version
+
+There are different UID2 token versions in current use, and later processing steps are a little different depending on whether the token version is v2 or a later version.
+
+You can check the token version using the first four characters of the token.
+
+To review detailed logic, see [UID2Encryption.cs, lines 36-50](https://github.com/IABTechLab/uid2-client-net/blob/6ac53b106301e431a4aada3cbfbb93f8164ff7be/src/UID2.Client/UID2Encryption.cs#L36-L50).
+
+### Decrypt the Token
+
+Use the master key and site key to decrypt the token. For a code example, refer to the `Decrypt` function: see [UID2Encryption.cs, line 29](https://github.com/IABTechLab/uid2-client-net/blob/6ac53b106301e431a4aada3cbfbb93f8164ff7be/src/UID2.Client/UID2Encryption.cs#L29). This function decrypts UID2 tokens into raw UID2s as part of the UID2 SDK for C# / .NET, and includes logic to handle different token versions.
+
+### Make Sure the Token Has Not Expired
+
+A token must be valid and current so that it can be used in the bidstream. You must make sure that it hasn't expired. To do this, check these two conditions:
+
+- The `remaining_lifetime` or `lifetime` value must not exceed the `max_bidstream_lifetime_seconds` value.
+- The `skew_duration` value must not exceed the `allow_clock_skew_seconds` value.
+
+ For an example of how this is done, review the code for the `DoesTokenHaveValidLifetimeImpl` function: see [UID2Encryption.cs, line 237](https://github.com/IABTechLab/uid2-client-net/blob/6ac53b106301e431a4aada3cbfbb93f8164ff7be/src/UID2.Client/UID2Encryption.cs#L237).
+
+The following sections show how the `lifetime` and `skew_duration` values for a token are calculated. The calculation is a little different depending on the token version:
+
+- [Calculating Token Expiration: Token v2](#calculating-token-expiration-token-v2)
+- [Calculating Token Expiration: All Later Versions](#calculating-token-expiration-all-later-versions)
+
+#### Calculating Token Expiration: Token v2
+
+For token v2, the calculation to make sure that the token hasn't expired is as follows:
+
+`remaining_lifetime` = **Token Expiry** &#8211; **Current Time**
+
+`skew_duration` = `0`
+
+For v2, we use the token expiry minus the current time to calculate the remaining lifetime. This is because v2 doesn't have a **Token Generated** field, which is present in later versions. All token versions have an **Identity Established** field, but this indicates the time that the original token was generated, before any token refreshes, so it can't be used to calculate whether the token is still valid.
+
+#### Calculating Token Expiration: All Later Versions
```

Comment:
we're also calculating the token lifetime here

Reply:
yep, done, thx.

Reply:
Updated to state both.

Reply:
As above: took this out again based on input from @caroline-ttd.